### PR TITLE
Generate map.dist.yml and set.dist.yml

### PR DESCRIPTION
**Pull request by foolip**
This requires a lot of overrides to get reasonable browser versions.
Firefox 25 seems like the right point to stop, but it's not objectively
so. For some uses of Map and Set, the forEach() method isn't needed.

Also simplify the spec URL list for Map.

### CODE REVIEW

**Review by Elchi3 — COMMENTED** [review]
As said when creating this data in https://github.com/web-platform-dx/web-features/pull/712#discussion_r1533453816, I would be much more comfortable with a property `compat_features_ignored_for_status` or similar instead of commenting things out. You've done a great job to provide reasons for each ignored feature here, maybe reasons should be required whenever adding to the ignore list.

What state do we call Maps when all the ignored featured are also available? This adjusts "Maps (initial support)" but maybe users are interested in "Maps (fully compliant)". I think the "(initial support)" remark becomes quite important now, because with all the ignored features, the baseline status here really is for the initial support of Maps not of fully compliant Maps with all tests passing etc.

**Comment by foolip** [comment]
@Elchi3 I agree we should exclude individual keys instead of overriding the list as a whole. See https://github.com/web-platform-dx/web-features/issues/841 for what I think that could look like.

**Comment by Elchi3** [comment]
In fact there are two reasons given here: "predating" and "non-essential".

Predating depends on how you look at it. If you look at snapshot data, all features listed as "predated" are part of ECMAScript 2015. Now ECMAScript 2015 was quite a large snapshot, so your predating is probably referring to the fact that most engines first implemented Maps without iterators and then iterators came shortly after. If someone wants the full ECMAScript 2015 Maps, then the following features marked as "predated" shouldn't be ignored, though. I originally added these in because my main data point were snapshots which are more neutral to me than what certain engines have implemented when.

```
# Map predates the iterable protocol.
# - javascript.builtins.Map.Map.iterable_allowed
# Map predates Symbol.iterator.
# - javascript.builtins.Map.@@iterator
# Map predates Symbol.species.
# - javascript.builtins.Map.@@species
```

"Non-essential" seems subjective but I tend to agree that in order to make use of Maps, you don't necessarily need to worry about these features. If you get bitten by this and the feature was declared baseline high to you (and thus not expecting bugs/interop issues), then that might lead to frustrations. In the case here everything seems to be supported long enough that such frustrations are unlikely today, though. 

```
# Support for `new Map(null)` isn't essential.
# - javascript.builtins.Map.Map.null_allowed
# Equality of 0 and -1 isn't essential for most uses of Map.
# - javascript.builtins.Map.key_equality_for_zeros
```

Overall it looks like both is done to move
- the Firefox version from Firefox 36/37/41 (41 is 2015-09-22) to Firefox 25 (2013-10-29)
- the Edge version from Edge 13 (2015-11-12) to Edge 12 (2015-07-29)
- the Chrome version from 43/51 (51 is 2016-05-25) to 38 (2014-10-07)
- the Safari version from 9/10 (10 is 2016-09-20) to 8 (2014-10-16)

With the effect that instead of 
- `baseline_low_date: 2016-09-20` (Safari 10)

we have a date that is 1 year and 2 months older:
- `baseline_low_date: 2015-07-29` (Edge 12)

Now all this stuff is from long time ago, so maybe it doesn't matter so much. For more recent features, it seems to me that deciding what to ignore and what not is quite important and I guess it would be helpful to have guidelines for it.

File: feature-group-definitions/map.dist.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    safari_ios: "8"
+# Overrides are applied to exclude many non-essential parts of Map.
+# The net effect is to bump the earliest Firefox version from 13 to
+# 25, which is when forEach() support was added.
+compat_features:
+  - javascript.builtins.Map
+  - javascript.builtins.Map.Map
+  # Map predates the iterable protocol.
+  # - javascript.builtins.Map.Map.iterable_allowed
+  # Support for `new new Map(null)` isn't essential.
```
Comment (Elchi3) [inline feature-group-definitions/map.dist.yml:29]:
```suggestion
  # Support for `new Map(null)` isn't essential.
```

**Comment by ddbeck** [comment]
> For the arrays group, I have a lot more features, one is array-iterators.yml. Maybe, instead of fiddling with ignores, map-iterators.yml and set-iterators.yml are needed here.

@foolip What do you think of @Elchi3's suggestion above? That feels like something that would unblock this and in this particular case, we don't have to stress about alignment with caniuse (because caniuse doesn't represent Map at all).

**Comment by foolip** [comment]
Since we already have groups for arrays, maps, and sets, I think that's a natural way to organize this, and I can't think of any serious downsides. But I think that "living groups" is not going to be a great fit for most of the web platform, and I've commented in https://github.com/web-platform-dx/web-features/issues/866#issuecomment-2072885726 with an approach I've been thinking about.

Here, even if we split out iterators, the issue of `javascript.builtins.Map.key_equality_for_zeros` remains, and that's the kind of case I have in mind for `later_support`.

I think the "last enhanced" part of @Elchi3's suggestion makes sense and maps onto living features, but I'm not sure that we should give groups any statuses. For most groups an aggregated status won't make any sense, and conversely a "last enhanced" property would make sense for a lot of existing features where we've had to exclude some minor additions to get the browser versions right.

**Comment by Elchi3** [comment]
> Here, even if we split out iterators, the issue of `javascript.builtins.Map.key_equality_for_zeros` remains, and that's the kind of case I have in mind for `later_support`.

interesting, I think `new_required` BCD entries are the same kind of thing and you decided to remove these entirely from BCD and web-features in https://github.com/web-platform-dx/web-features/pull/871. So, to me the other option for `javascript.builtins.Map.key_equality_for_zeros` would be data removal, too. The `later_support` concept would work too probably, but I think I'm now puzzled when to remove and when to use `later_support`.

**Comment by Elchi3** [comment]
> a "last enhanced" property would make sense for a lot of existing features where we've had to exclude some minor additions to get the browser versions right.

+1 to this

**Comment by foolip** [comment]
> > Here, even if we split out iterators, the issue of `javascript.builtins.Map.key_equality_for_zeros` remains, and that's the kind of case I have in mind for `later_support`.
> 
> interesting, I think `new_required` BCD entries are the same kind of thing and you decided to remove these entirely from BCD and web-features in #871. So, to me the other option for `javascript.builtins.Map.key_equality_for_zeros` would be data removal, too. The `later_support` concept would work too probably, but I think I'm now puzzled when to remove and when to use `later_support`.

The litmus test that I would apply is which direction compatibility goes in. If there is code that can be written against current browsers that wouldn't work in older browsers, then I think that warrants a BCD entry describing the new behavior and which browser releases added support for it. But when the old behavior is the more permissive or "larger" API and it was restricted, then there is no code that works in new browsers that would break in older ones. Such entries only make sense temporarily to communicate a breaking change, and can be removed after some time.

I think `javascript.builtins.Map.key_equality_for_zeros` falls into the first category, since you can write code assuming that positive and negative zero are the same key which works today, but didn't work originally.

**Comment by ddbeck** [comment]
> Such entries only make sense temporarily to communicate a breaking change, and can be removed after some time.

I think a good approach for these things in BCD is to flip the compat story. Describe the old behavior as a non-standard, deprecated "feature" instead of the new behavior as a permanently relevant feature.

For example, instead of `javascript.builtins.Map.Map.null_allowed` you'd have something like `javascript.builtins.Map.Map.null_throws`, with a non-standard, deprecated status. Two years after the last browser "unships" this "feature", you drop it from BCD because it meets the irrelevance guideline.

You can do similar things for stuff like callbacks that got turned into promises.

**Comment by foolip** [comment]
These dist files were added in https://github.com/web-platform-dx/web-features/pull/1227 but things are quite a bit different now that we have `compute_from` and statuses per BCD key.

**Pull request closed without merging**